cmake/FindHIP.cmake

```diff
@@ -370,6 +391,10 @@ macro(HIP_PREPARE_TARGET_COMMANDS _target _format _generated_files _source_files
     # Initialize list of includes with those specified by the user. Append with
     # ones specified to cmake directly.
     set(HIP_HIPCC_INCLUDE_ARGS ${HIP_HIPCC_INCLUDE_ARGS_USER})
+
+    # Add the include directories
+    list(APPEND HIP_HIPCC_INCLUDE_ARGS "-I$<JOIN:$<TARGET_PROPERTY:${_target},INCLUDE_DIRECTORIES>, -I>")
```

Comment:
I didn't notice this before. The expression assumes that ```TARGET_PROPERTY:INCLUDE_DIRECTORIES``` is non-empty. It needs to be checked for non-empty, else a dangling ```-I``` would be appended to ```HIP_HIPCC_INCLUDE_ARGS```. See https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#output-expressions for how to do that.